src/drivers/utils/index.ts

```diff
@@ -1,11 +1,13 @@
 import type { Driver } from "../..";
 
-type DriverFactory<T> = (opts: T) => Driver;
+type DriverFactory<OptionsT, InstanceT> = (
+  opts: OptionsT
+) => Driver<OptionsT, InstanceT>;
 interface ErrorOptions {}
 
-export function defineDriver<T = any>(
-  factory: DriverFactory<T>
-): DriverFactory<T> {
+export function defineDriver<OptionsT = any, InstanceT = any>(
```

Comment:
This way, default instance is not specified unless typed wdyt? 

```suggestion
export function defineDriver<OptionsT = any, InstanceT = never>(
```

Reply:
had to revert in 256c1923653c5cc76ebc258b667e609503f867e4